app/controller/InfoController.js

```diff
@@ -409,22 +440,107 @@ SDL.InfoController = Em.Object.create(
       });
     },
 
-    extractEntrypointFromManifest: function(bundle_path) {
+    /**
+     * @description Sends request to backend to read manifest.js for a specified app
+     * @param {String} policyAppID Id of application to read manifest
+     */
+    getWebAppManifestContent: function(policyAppID) {
       return new Promise( (resolve, reject) => {
-        const manifest_path = bundle_path + 'manifest.js';
+        let manifest_path =
+          `${SDL.InfoController.getWebEngineOutputFolder()}${policyAppID}/manifest.js`;
+        let client = FFW.RPCSimpleClient;
 
-        Em.Logger.log(`App store: importing manifest from ${manifest_path}`);
-        import(manifest_path)
-          .then((content) => {
-            if ('entrypoint' in content) {
-              resolve(content['entrypoint']);
-            }
+        let manifest_receive_timer = setTimeout(function() {
+          Em.Logger.log('App store: Timeout for getting manifest expired');
+          client.unsubscribeFromEvent('GetAppManifestResponse');
+          reject();
+        }, 10000);
 
+        let manifest_received_callback = function(params) {
+          Em.Logger.log('App store: Manifest loading has finished');
+          clearTimeout(manifest_receive_timer);
+          client.unsubscribeFromEvent('GetAppManifestResponse');
+
+          if (params.success == false) {
+            Em.Logger.log('App store: Manifest loading was not successful');
             reject();
-        });
+          }
+
+          Em.Logger.log('App store: Manifest was loaded successfully');
+          resolve(params['content']);
+        }
+
+        let message = {
+          method: 'GetAppManifestRequest',
+          params: {
+            'fileName': manifest_path
+          }
+        };
+
+        Em.Logger.log(`App store: Loading manifest for ${policyAppID}`);
+        client.connect();
+        client.subscribeOnEvent('GetAppManifestResponse', manifest_received_callback);
+        client.send(message);
       });
     },
 
+    /**
+     * @description Extracts entrypoint name from manifest content
+     * @param {String} bundle_manifest stringified content of app manifest
+     */
+    extractEntryPointFromManifest: function(bundle_manifest) {
+      return new Promise( (resolve, reject) => {
+        Em.Logger.log(`App store: parsing manifest content`);
+
+        var bundle_json;
+        try {
+          bundle_json = JSON.parse(bundle_manifest)
+        }
+        catch {
+          Em.Logger.log(`App store: failed to parse JSON content`);
+          reject();
+        }
+
+        Em.Logger.log(`App store: manifest parsed successfully`);
+        if (!'entrypoint' in bundle_json) {

```

Comment:
```suggestion
        if (!('entrypoint' in bundle_json)) {
```

order of operations, it will evaluate `!'entrypoint'` to false and then do `false in bundle_json`

Reply:
@iCollin thanks, fixed in https://github.com/smartdevicelink/sdl_hmi/pull/274/commits/b506afc2d807d2ffa04860513272c8f6129f7f03